app/src/main/java/org/mozilla/fenix/library/history/HistoryView.kt

```diff
@@ -31,8 +34,20 @@ class HistoryView(
     var mode: HistoryFragmentState.Mode = HistoryFragmentState.Mode.Normal
         private set
 
-    val historyAdapter = HistoryAdapter(interactor)
+    val historyAdapter = HistoryAdapter(interactor).apply {
+        addLoadStateListener {
+            adapterItemCount = itemCount
+            Log.d("hehehaha", "adapterItemCount = $adapterItemCount")
```

Comment:
😛

Reply:
thanks for pointing that out! changing getHistory() method to suspend changed the order of update() calls. For some reasons tests were not affected